src/utils/isReactComponentClass.js

```diff
@@ -44,6 +44,26 @@ export default function isReactComponentClass(
     return true;
   }
 
+  // check for @extends React.Component in docblock
+  if (path.parentPath.value && path.parentPath.value) {
+    var classDeclaration = {};
+    if (Array.isArray(path.parentPath.value)) {
+      var matches = path.parentPath.value.filter(function(declaration) { return declaration.type === 'ClassDeclaration' });
+      if (matches[0]) {
+        classDeclaration = matches[0];
+      }
+    } else {
+      classDeclaration = path.parentPath.value;
+    }
+    
+    if (classDeclaration.leadingComments && classDeclaration.leadingComments.length > 0) {
+      var matchedComments = classDeclaration.leadingComments.filter(function(comment) { return comment.value.match(/(@extends React.Component)/) });
```

Comment:
A few issues:

1. No need to wrap the inside of the `RegExp` with parentheses, since you're not capturing anything
2. The `.` character in your `RegExp` will match anything -- you need to escape it to check for a literal period. Also, some people might indent the JSDoc field and value differently, so best to be lenient for that:

```js
/@extends\s+React\.Component/
```

3. No need to check through all of the comment lines -- you just need to find one that matches. So you can use `Array.prototype.some` to your advantage:

```js
if (classDeclaration.leadingComments.some(function (comment) {
  return /@extends\s+React\.Component/.test(comment.value);
})) {
  return true;
}
```

Reply:
Excellent points, getting all of that integrated.